### PR TITLE
feat(resharding): set defaults for state split config

### PR DESCRIPTION
**Pull request by wacban**
Without the individual defaults for fields it was impossible to set only some fields while leaving others out. Now by setting only some fields the rest will get the sensible defaults. It was annoying for mocknet and would be for node maintainers if they need to adjust this config.

### CODE REVIEW

File: core/chain-configs/src/client_config.rs [hunk 1]
```diff
@@ -167,40 +167,65 @@ pub struct StateSplitConfig {
     /// The soft limit on the size of a single batch. The batch size can be
     /// decreased if resharding is consuming too many resources and interfering
     /// with regular node operation.
+    #[serde(default = "default_batch_size")]
```
Comment (shreyan-gupta) [inline core/chain-configs/src/client_config.rs:170]: I'm wondering if we can use the serde default attribute for structs. I haven't tested it but it's behavior should be different from the serde default attribute of fields, where for structs, we take the field default value from Default::default()

Documentation here: https://serde.rs/container-attrs.html#default

Comment (wacban) [reply core/chain-configs/src/client_config.rs:170]: Let me test it out.

**Comment by shreyan-gupta** [comment]
FYI, this works! :)
```
#[derive(Serialize, Deserialize, Debug)]
#[serde(default)]
struct Point {
    x: i32,
    y: i32,
}

impl Default for Point {
    fn default() -> Self {
        Self { x: 1, y: 2 }
    }
}

#[test]
fn serde() {
    let point = Point { x: 10, y: 20 };
    let serialized = serde_json::to_string(&point).unwrap();
    println!("serialized = {}", serialized);
    let deserialized: Point = serde_json::from_str(&serialized).unwrap();
    println!("deserialized = {:?}", deserialized); // I get Point { x: 10, y: 20 }
    let serialized = "{\"x\":15}";
    let deserialized: Point = serde_json::from_str(&serialized).unwrap();
    println!("deserialized = {:?}", deserialized); // I get Point { x: 15, y: 2 }
}
```

File: core/chain-configs/src/client_config.rs (hunk 1 above)
Comment (shreyan-gupta) [reply core/chain-configs/src/client_config.rs:170]: Just posted in comment, good timings

**Review by shreyan-gupta — APPROVED** [review]

**Pull request merged**